code-studio/src/js/components/pairing/student_selector.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { studentsShape } from './types';
+
+/**
+ * A component for selecting one or more students in a section.
+ */
+const StudentSelector = React.createClass({
+  propTypes: {
+    students: studentsShape,
+    handleSubmit: React.PropTypes.func.isRequired
+  },
+
+  getInitialState() {
+    return {
+      selectedStudentIds: []
+    };
+  },
+
+  handleStudentClicked(event) {
+    var selectedStudentIds = this.state.selectedStudentIds;
+    var studentId = +event.target.getAttribute('data-id');
+    var index = selectedStudentIds.indexOf(studentId);
+    if (index === -1) {
+      // not selected, select it
+      selectedStudentIds.push(studentId);
+    } else {
+      // selected, unselect it
+      selectedStudentIds.splice(index, 1);
+    }
+    this.setState({selectedStudentIds: selectedStudentIds});
+  },
+
+  handleSubmit(event) {
+    this.props.handleSubmit(this.state.selectedStudentIds);
+    event.preventDefault();
+  },
+
+  render() {
+    if (!this.props.students) {
+      return null;
+    } else if (this.props.students.length === 0) {
+      return <span>There are no other students in this section.</span>;
+    }
+
+    var studentDivs = this.props.students.map(student => {
+      var className = "student selectable";
+      if (this.state.selectedStudentIds.indexOf(student.id) !== -1) {
+        className = "student selectable selected";
```

Comment:
you can use [classnames](https://github.com/JedWatson/classnames) to make this a lot cleaner:
```javascript
let selected = this.state.selectedStudentIds.indexOf(student.id) !== -1;
let className = classNames("student", "selectable", { selected });
```